TASK: Show tooltips instead of default browser titles

aertmann opened the pull request:
In many cases the icon is not self explanatory and we rely on a title being shown on hover.
The tooltip styling is improved and tooltips are improved in general.

Tooltips have been inserted on almost every iconic button throughout
the editing interface as well as in the other modules.

NEOS-1122 #close

aertmann commented:
Partly migrated from https://review.typo3.org/#/c/38120/ and https://github.com/patrickreck/neos-development-collection/commits/gerrit-38120 – thanks to @patrickreck and @jaguerra

aertmann commented:
![image](https://cloud.githubusercontent.com/assets/903567/14285205/03df3af8-fb8e-11e5-9878-190075cfc4b0.png)
![image](https://cloud.githubusercontent.com/assets/903567/14285222/0b90dae0-fb8e-11e5-8471-9dcdb924012d.png)
![image](https://cloud.githubusercontent.com/assets/903567/14285233/11a31a24-fb8e-11e5-8e2d-91ce8fc4f3c8.png)
![image](https://cloud.githubusercontent.com/assets/903567/14285244/1a8743e0-fb8e-11e5-950e-1c2d9e591371.png)
![image](https://cloud.githubusercontent.com/assets/903567/14285253/23f5f41c-fb8e-11e5-83d5-1df1b478772e.png)
![image](https://cloud.githubusercontent.com/assets/903567/14285262/2c507326-fb8e-11e5-9432-17e3882e195d.png)
![image](https://cloud.githubusercontent.com/assets/903567/14285269/2f52739e-fb8e-11e5-9de2-0d3463ff6869.png)
![image](https://cloud.githubusercontent.com/assets/903567/14285274/31cf71c6-fb8e-11e5-9969-ce442e8011a7.png)
![image](https://cloud.githubusercontent.com/assets/903567/14285282/38f76210-fb8e-11e5-9472-d0bfcaca6b18.png)
![image](https://cloud.githubusercontent.com/assets/903567/14285287/3e5718cc-fb8e-11e5-90b3-6ff3e9161124.png)
![image](https://cloud.githubusercontent.com/assets/903567/14285292/416e4094-fb8e-11e5-8891-7e455fc6dbd0.png)
![image](https://cloud.githubusercontent.com/assets/903567/14285307/4ee07f9e-fb8e-11e5-8e6e-e32a5cecf4ea.png)
![image](https://cloud.githubusercontent.com/assets/903567/14285320/596ffb38-fb8e-11e5-96b1-af14969b54e0.png)
![image](https://cloud.githubusercontent.com/assets/903567/14285323/5d61fb92-fb8e-11e5-864e-e508d34407ae.png)
![image](https://cloud.githubusercontent.com/assets/903567/14285329/63d2ed4c-fb8e-11e5-84ce-36b4681d1b72.png)
![image](https://cloud.githubusercontent.com/assets/903567/14285333/6ad1c00a-fb8e-11e5-9cc3-9dc79b892ca3.png)
![image](https://cloud.githubusercontent.com/assets/903567/14285339/6efade46-fb8e-11e5-949b-8f8f69651345.png)
![image](https://cloud.githubusercontent.com/assets/903567/14285352/79d5892e-fb8e-11e5-88bf-b5f75a04fb93.png)
![image](https://cloud.githubusercontent.com/assets/903567/14285356/7caeeece-fb8e-11e5-8b56-dfce127c8283.png)
![image](https://cloud.githubusercontent.com/assets/903567/14285364/888481fa-fb8e-11e5-9479-a17eb5958f92.png)
![image](https://cloud.githubusercontent.com/assets/903567/14285376/95070c68-fb8e-11e5-994c-128c58c2f180.png)
![image](https://cloud.githubusercontent.com/assets/903567/14285378/97a075ea-fb8e-11e5-8bc3-2304a2b1395e.png)
![image](https://cloud.githubusercontent.com/assets/903567/14285388/a122412a-fb8e-11e5-9d75-36df952058e3.png)
![image](https://cloud.githubusercontent.com/assets/903567/14285395/a70376fe-fb8e-11e5-8037-b38b432fd264.png)
![image](https://cloud.githubusercontent.com/assets/903567/14285424/c89a5e4a-fb8e-11e5-9db4-388b8669c0f8.png)
![image](https://cloud.githubusercontent.com/assets/903567/14285426/cd183e9c-fb8e-11e5-8dcd-389bd652b4a4.png)
![image](https://cloud.githubusercontent.com/assets/903567/14285444/e2e61000-fb8e-11e5-9dc3-d8755457d637.png)

johannessteu commented:
Tested - works like a charm!

gerhard-boden commented:
Nice usability improvement! I especially like the description text for modules within the menu.

Found a few missing "standard" tooltips. Not sure if the plan is to remove all of them, or just the most important ones. It also could be that something went wrong on my end while building the CSS files.

![image](https://cloud.githubusercontent.com/assets/10533739/14300783/152f1896-fb93-11e5-95c6-83ebb80dd94a.png)
(only `Create new` has the fancy new tooltips in the content handlebar)

![image](https://cloud.githubusercontent.com/assets/10533739/14300965/3aa00f58-fb94-11e5-9ac8-954a294d40fa.png)
("User Settings" shows the standard tooltip too)

The positioning looks good in the following browsers (Win 10):
Firefox 43.0.4 
Chrome 49.0.2623.110
IE 11.0.29

aertmann commented:
Thanks for the quick and thorough review guys! Much appreciated.

Fixed the two issues @johannessteu mentioned, corrected a spelling mistake and changed "Remove" to "Delete" in the inline editing handles so it's in sync with the tree toolbars.

gerhard-boden commented:
You're most welcome! Just retested and tooltips look good and work :+1: 

It's easy for me to help out with UI testing, because I don't have to become acquainted with large portions of code first. So whenever you have something UI related to test, just ping me. Most of the time I will be able to test it within 24 hours.

gerhard-boden commented on TYPO3.Neos/Resources/Public/JavaScript/Content/Navigate/NodeTree.js:
```diff
@@ -226,6 +226,18 @@ define(
+						var nodeTypeLabel = I18n.translate(node.data.nodeTypeLabel),
+							tooltip = node.data.title;
+
+						if (nodeTypeLabel !== '' && tooltip.indexOf(nodeTypeLabel) === -1) {
```
this currently breaks master since `tooltip` seems to be `null`.
`ContentModule-built.js?bust=f0da31ac2bd0:422 Uncaught TypeError: Cannot read property 'indexOf' of null`

gerhard-boden commented:
Have to take back my +1. left a comment in the code. This change breaks master:
You can not add new nodes to the node tree.

Sorry, only checked if the tooltips where displayed correctly across browsers, but did not test for general backend functionality.

gerhard-boden commented:
just noticed that sometimes the label is stuck when moving nodes in the tree. It does not go away until refresh
![image](https://cloud.githubusercontent.com/assets/10533739/14355911/fd3d607a-fce2-11e5-8605-6e77bde067c4.png)

aertmann commented:
@gerhard-boden: Hey no backsies! Anyway sorry about that. Fixed it with #437 😬

aertmann commented:
fix for stuck tooltips https://github.com/neos/neos-development-collection/pull/444
